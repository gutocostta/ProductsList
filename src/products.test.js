import React from 'react';
import renderer from 'react-test-renderer';
import Products from './products';

const PROD_LIST  = [
    {
        name: 'Product 1',
        imageUrl: 'prod1.jpg',
        value: 10
    },
    {
        name: 'Product 2',
        imageUrl: 'prod2.jpg',
        value: 20
    },
    {
        name: 'Product 3',
        imageUrl: 'prod3.jpg',
        value: 30
    },
];

test('teste 1', () => {
    const component = renderer.create(
        <Products  />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('teste 2', async () => {
    window.fetch = jest.fn(async () => {
        return {
            json: async () => {
                return PROD_LIST
            }
        }
    });

    const component = renderer.create(
        <Products  />
    );

    setTimeout(function() {
        expect(window.fetch).toHaveBeenCalledWith('https://sleepy-argument.glitch.me/products');
    
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    },1000);

});