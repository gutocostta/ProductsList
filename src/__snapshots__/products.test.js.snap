// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  style={
    Object {
      "background-color": "fucsia",
      "border": "3px solid #00f9ff",
      "width": "800px",
    }
  }
>
  <h1>
    Products
  </h1>
  <ul />
  <a
    href="/"
  >
    Add Product
  </a>
</div>
`;
